app/iot/service.py

```diff
@@ -18,32 +18,54 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
+        attempts_left = 7
+        while device_id in self.devices:
+            print(
+                f"Device with id {device_id} already registered"
+                f"Gererating new one..."
+            )
+            device_id = generate_id()
+            attempts_left -= 1
+            if attempts_left == 0:
+                print("Unable to register device, trying again...")
+            print(f"Attempts left: {attempts_left}")
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
-        self.devices[device_id].disconnect()
-        del self.devices[device_id]
+    async def unregister_device(self, device_id: str) -> None:
+        try:
+            self.devices[device_id].disconnect()
+            del self.devices[device_id]
+        except KeyError:
+            print(f"Device {device_id} not found")
 
-    def get_device(self, device_id: str) -> Device:
-        return self.devices[device_id]
 
-    def run_program(self, program: list[Message]) -> None:
+    async def get_device(self, device_id: str) -> Device:
+        try:
+            return self.devices.get(device_id)
```

Comment:
Bug: The get method of a dictionary returns None if the key is not found. Therefore, this line will not raise an AttributeError. Instead, you should check if the returned value is None and handle the error accordingly.